vernac/declare.ml

```diff
@@ -734,14 +744,20 @@ let mutual_make_bodies ~typing_flags ~fixitems ~rec_declaration ~possible_indexe
     let vars = Vars.universes_of_constr (List.hd fixdecls) in
     vars, fixdecls, None
 
-let declare_mutually_recursive ~info ~cinfo ~opaque ~uctx ~rec_declaration ~possible_indexes =
+let declare_mutually_recursive ~info ~cinfo ~opaque ~uctx ~rec_declaration ~possible_indexes ?using () =
   let { Info.poly; udecl; scope; clearbody; kind; typing_flags; user_warns; ntns; _ } = info in
+  let env = Global.env() in
   let vars, fixdecls, indexes =
-    mutual_make_bodies ~typing_flags ~fixitems:cinfo ~rec_declaration ~possible_indexes in
+    mutual_make_bodies env ~typing_flags ~fixitems:cinfo ~rec_declaration ~possible_indexes in
   let uctx = UState.restrict uctx vars in
   let univs = UState.check_univ_decl ~poly uctx udecl in
+  let evd = Evd.from_env env in
+  (* let using = Option.map (interp_proof_using_gen f env evd [name, typ, body]) using in *)
   let csts = CList.map2
-      (fun CInfo.{ name; typ; impargs; using } body ->
+      (fun CInfo.{ name; typ; impargs } body ->
+         let f (name, typ, body) =
+           name, List.map EConstr.of_constr [typ; body] in
+         let using = Option.map (interp_proof_using_gen f env evd [name, typ, body]) using in
```

Comment:
It would not change the final behavior, but elsewhere, the order is `[name, body, typ]`.

Reply:
I think this is how it was done before, so I tried to preserve it. Before we were not consistent. Happy to pick a consistent order.

Reply:
OK